src/main/java/useCases/generator/DestinyCardPoolGenerator.java

```diff
@@ -17,74 +14,44 @@
-     * Each line in the file represents a single destiny card with its associated actions.
-     * The destiny cards are created and added to the card pool, which is then added to the given Destiny instance.
-     * Destiny cards are used in the game to determine various outcomes and events.
+     * Generates a pool of DestinyCards and adds them to the specified Destiny instance.
      *
-     * @param fileName The name of the information file containing destiny card data.
-     * @param destiny  The Destiny instance to which the generated card pool will be added.
+     * @param fileName The name of the CSV file containing DestinyCard information.
+     * @param destiny  The Destiny instance to which the generated DestinyCards will be added.
+     * @throws NullPointerException     If the provided Destiny instance is null.
+     * @throws NumberFormatException    If action values in the CSV file are not in the expected numeric format.
      */
-    public static void generateDestinyCardPool(String fileName, Destiny destiny){
+    public static void generateDestinyCardPool(String fileName, Destiny destiny) {
         if (destiny == null) {
             throw new NullPointerException("The Destiny instance must not be null.");
         }
 
         ArrayList<DestinyCard> cardPool = new ArrayList<>();
-        try (BufferedReader br = new BufferedReader(new FileReader(fileName))){
-            String line;
-            ArrayList<Object> actions = new ArrayList<>(Arrays.asList(0, 0, 0));
-            while ((line = br.readLine()) != null){
-                String[] values = line.split(DELIMITER);
-
-                if (values.length < 4) {
-                    // Handle the case where there are missing values in the CSV line
-                    throw new IllegalArgumentException("Invalid data in the CSV file: " + line);
-                }
-
-                // Generate the actions
-                try {
-                    actions.set(0, Integer.parseInt(values[1]));
-                    actions.set(1, Integer.parseInt(values[2]));
-                    actions.set(2, Integer.parseInt(values[3]));
-                } catch (NumberFormatException e) {
-                    // Handle the case where action values are not in the expected numeric format
-                    throw new NumberFormatException("Invalid action value in the CSV file: " + line);
-                }
-
-                // Remove the "" in the content
-                String message = values[0].replace("\"", "");
-
-                DestinyCard card = new DestinyCard(message, actions);
-                cardPool.add(card);
+        ArrayList<Object> actions = new ArrayList<>(Arrays.asList(0, 0, 0));
+
+        DestinyCardDataManager dataManager = new DestinyCardDataManager();
+        ArrayList<String[]> data = dataManager.readData(fileName);
+
+        for(String[] values: data){
+            try {
+                actions.set(0, Integer.parseInt(values[1]));
```

Comment:
Maybe we can try adding some documentation here, so that we know what values[1], values[2], and values[3] mean. 

I think it means money gained, blocks moved, and rounds blocked, correct?